Source/SwiftLintFramework/Rules/Lint/DiscouragedDefaultSwitchCaseRule.swift

```diff
@@ -0,0 +1,190 @@
+        nonTriggeringExamples: [
+            Example(
+                #"""
+                enum StateOfMatter {
+                    case liquid
+                    case solid
+                    case gas
+                    case plasm
+
+                    var description: String {
+                        switch self {
+                            case .liquid: return "Liquid"
+                            case .solid: return "Solid"
+                            case .gas: return "Gas"
+                            case .plasm: return "Plasm"
+                        }
+                    }
+                }
+                """#
+            ),
+            Example(
+                #"""
+                switch number {
+                    case 1...4: return "Number is less than 5"
+                    case 5: return "Number is 5"
+                    default: return "Number is greater than 5"
+                }
+                """#
+            ),
+            Example(
+                #"""
+                switch text {
+                    case "Welcome": return true
+                    case "Goodbye": return false
+                    default: return nil
+                }
+                """#
+            ),
+            Example(
+                #"""
+                switch value {
+                    case .one: return "One"
+                    case .two: return "Two"
+                    case .default: return "Default"
+                }
+                """#
+            ),
+            Example(
+                #"""
+                switch result {
+                    case .success(let value): print(value)
+                    case .failure(let error): print(error)
+                }
+                """#
+            ),
+            Example(
+                #"""
+                switch (status, value) {
+                    case (.on, .green): return true
+                    default: return false
+                }
+                """#
+            )
+        ],
+        triggeringExamples: [
+            Example(
+                #"""
+                enum StateOfMatter {
+                    case liquid
+                    case solid
+                    case gas
+                    case plasm
+
+                    var description: String {
+                        switch self {
+                            case .liquid: return "Liquid"
+                            case .solid: return "Solid"
+                            ↓default: return "Other stuff"
+                        }
+                    }
+                }
+                """#
+            ),
+            Example(
+                #"""
+                switch CLLocationManager.authorizationStatus() {
+                    case .authorizedAlways, .authorizedWhenInUse, .notDetermined: return "authorized"
+                    case .denied, .restricted: return "denied"
+                    @unknown ↓default: fatalError("Unhandled case!")
```

Comment:
hrm, what's the resolution here though?

Reply:
Keith has a good point. I think these shouldn't trigger. What do you think @otaviocc ?

Reply:
@keith, @jpsim Yep, it shouldn't. I'll work on it (as soon as I find a way to check if it's `unknown`. I'm gonna work on it tomorrow. Thanks!